astropy/modeling/tests/test_fitters.py

```diff
@@ -654,6 +654,82 @@ def test_with_fitters_and_sigma_clip(self):
                         atol=1e-1)
 
 
+@pytest.mark.skipif('not HAS_SCIPY')
+class TestWeightedFittingWithOutlierRemoval:
+    """Issue #7020 """
+
+    def setup_class(self):
+        # values of x,y not important as we fit y(x,y) = p0 model here
+        self.y, self.x = np.mgrid[0:20, 0:20]
+        self.z = np.mod(self.x + self.y, 2) * 2 - 1 # -1,1 chessboard
+        self.weights = np.mod(self.x + self.y, 2) * 2 + 1 # 1,3 chessboard
+        self.z[0,0] = 1000.0 # outlier
+        self.z[0,1] = 1000.0 # outlier
+        self.x1d = self.x.flatten()
+        self.z1d = self.z.flatten()
+        self.weights1d = self.weights.flatten()
+
+    def test_1d_without_weights_without_sigma_clip(self):
+        model = models.Polynomial1D(0)
```

Comment:
this line is repeated throughout the test methods. It would be better to move it inside `setup_class` with a more descriptive name, like ``poly_1d = models.Polynomial1D(0)``